subsys/net/openthread/rpc/client/ot_rpc_udp_client.c

```diff
@@ -0,0 +1,247 @@
+
+	if (!zcbor_uint_encode(ctx.zs, &key, sizeof(key))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	if (!zcbor_bstr_encode_ptr(ctx.zs, (const char *)&aSockName->mAddress.mFields.m8,
+				   sizeof(aSockName->mAddress.mFields.m8))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	if (!zcbor_uint_encode(ctx.zs, (const char *)&aSockName->mPort, sizeof(aSockName->mPort))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_UDP_CONNECT, &ctx, ot_rpc_decode_error,
+				&error);
+
+exit:
+	return error;
+}
+
+otError otUdpClose(otInstance *aInstance, otUdpSocket *aSocket)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	ot_socket_key key = (ot_socket_key)aSocket;
+	otError error = OT_ERROR_NONE;
+
+	ARG_UNUSED(aInstance);
+
+	if (aSocket == NULL) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, sizeof(key) + 2);
+
+	if (!zcbor_uint_encode(ctx.zs, &key, sizeof(key))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_UDP_CLOSE, &ctx, ot_rpc_decode_error, &error);
+
+exit:
+	return error;
+}
+
+otError otUdpBind(otInstance *aInstance, otUdpSocket *aSocket, const otSockAddr *aSockName,
+		  otNetifIdentifier aNetif)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	ot_socket_key key = (ot_socket_key)aSocket;
+	otError error = OT_ERROR_NONE;
+	uint32_t netIf = aNetif;
+
+	ARG_UNUSED(aInstance);
+
+	if (aSocket == NULL) {
+		return OT_ERROR_INVALID_ARGS;
+	}
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, sizeof(key) + sizeof(otSockAddr) + sizeof(uint32_t) + 6);
+
+	if (!zcbor_uint_encode(ctx.zs, &key, sizeof(key))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	if (!zcbor_bstr_encode_ptr(ctx.zs, (const char *)&aSockName->mAddress.mFields.m8,
+				   sizeof(aSockName->mAddress.mFields.m8))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	if (!zcbor_uint_encode(ctx.zs, (const char *)&aSockName->mPort, sizeof(aSockName->mPort))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	if (!zcbor_uint_encode(ctx.zs, &netIf, sizeof(netIf))) {
+		NRF_RPC_CBOR_DISCARD(&ot_group, ctx);
+		error = OT_ERROR_INVALID_ARGS;
+		goto exit;
+	}
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_UDP_BIND, &ctx, ot_rpc_decode_error, &error);
+
+exit:
+	return error;
+}
+
+static void ot_rpc_cmd_udp_receive_cb(const struct nrf_rpc_group *group,
+				      struct nrf_rpc_cbor_ctx *ctx, void *handler_data)
+{
+	bool decoding_ok;
+	otMessageInfo message_info;
+	ot_msg_key msg_key = 0;
+	ot_socket_key soc_key = 0;
+	otUdpSocket *socket;
+
+	decoding_ok = zcbor_uint_decode(ctx->zs, &soc_key, sizeof(soc_key));
```

Comment:
I guess it is much optimal to use decoders from `nrf_rpc_serializer.h` and finally `nrf_rpc_decoding_done_and_check`.
However, it is possible to do as part of `Polishing` task.